templates/types/streaming/nextjs/app/api/chat/embed/embeddings.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Document,
+  IngestionPipeline,
+  Settings,
+  SimpleNodeParser,
+} from "llamaindex";
+import { DocxReader } from "llamaindex/readers/DocxReader";
+import { PDFReader } from "llamaindex/readers/PDFReader";
+import { TextFileReader } from "llamaindex/readers/TextFileReader";
+
+export async function uploadDocument(raw: string): Promise<string[]> {
+  const [header, content] = raw.split(",");
+  const mimeType = header.replace("data:", "").replace(";base64", "");
+  const fileBuffer = Buffer.from(content, "base64");
+  const documents = await loadDocuments(fileBuffer, mimeType);
+  return await runPipeline(documents);
+}
+
+async function runPipeline(documents: Document[]): Promise<string[]> {
+  // mark documents to add to the vector store as private
+  for (const document of documents) {
+    document.metadata = {
+      ...document.metadata,
+      private: true,
+    };
+  }
+  const pipeline = new IngestionPipeline({
+    transformations: [
+      new SimpleNodeParser({
+        chunkSize: Settings.chunkSize,
+        chunkOverlap: Settings.chunkOverlap,
+      }),
+      Settings.embedModel,
+    ],
+    // TODO: make sure this adds the nodes to the vector DB
```

Comment:
@leehuwuj here the user uploads documents and they should be added to the vector db